cfme/storage/volume_snapshot.py

```diff
@@ -0,0 +1,238 @@
+        'contains(@class, "flash_text_div")]')
+
+    @property
+    def in_volume_snapshot(self):
+        return (
+            self.logged_in_as_current_user and
+            self.navigation.currently_selected == ['Storage', 'Block Storage', 'Volume Snapshots']
+        )
+
+    @property
+    def is_displayed(self):
+        return self.in_volume_snapshot
+
+
+class VolumeSnapshotAllView(VolumeSnapshotView):
+    """The all Volume Snapshot page"""
+    toolbar = View.nested(VolumeSnapshotToolbar)
+    including_entities = View.include(BaseEntitiesView, use_parent=True)
+
+    @property
+    def is_displayed(self):
+        return (
+            self.in_volume_snapshot and
+            self.title.text == 'Cloud Volume Snapshots')
+
+
+class VolumeSnapshotDetailsView(VolumeSnapshotView):
+    """The detail Volume Snapshot page"""
+    @property
+    def is_displayed(self):
+        expected_title = '{} (Summary)'.format(self.context['object'].name)
+
+        return (
+            self.in_volume_snapshot and
+            self.title.text == expected_title and
+            self.entities.breadcrumb.active_location == expected_title)
+
+    toolbar = View.nested(VolumeSnapshotDetailsToolbar)
+    sidebar = View.nested(VolumeSnapshotDetailSidebar)
+    entities = View.nested(VolumeSnapshotDetailsEntities)
+
+
+@attr.s
+class VolumeSnapshot(BaseEntity, WidgetasticTaggable):
+    """ Model of an Storage Volume Snapshots in cfme
+
+    Args:
+        name: name of the snapshot
+        provider: provider
+    """
+    name = attr.ib()
+    provider = attr.ib()
+
+    def refresh(self):
+        self.provider.refresh_provider_relationships()
+        self.browser.refresh()
+
+    @property
+    def exists(self):
+        """ check for snapshot exist on UI.
+
+        Returns:
+            :py:class:`bool'
+        """
+        view = navigate_to(self.parent, 'All')
+        return self.name in view.entities.all_entity_names
+
+    @property
+    def status(self):
+        """ status of cloud volume snapshot.
+
+        Returns:
+            :py:class:`str' Status of volume snapshot.
+        """
+        view = navigate_to(self.parent, 'All')
+        view.toolbar.view_selector.select("List View")
+
+        for item in view.entities.elements.read():
+            if self.name in item['Name']:
+                return str(item['Status'])
+
+    @property
+    def size(self):
+        """ size of cloud volume snapshot.
+
+        Returns:
+            :py:class:`int' size of volume snapshot in GB.
+        """
+        view = navigate_to(self, 'Details')
+        return int(view.entities.properties.get_text_of('Size').split()[0])
+
+    @property
+    def volume(self):
+        """ volume name of snapshot.
+
+        Returns:
+            :py:class:`str' respective volume name.
+        """
+        view = navigate_to(self, 'Details')
+        return view.entities.relationships.get_text_of('Cloud Volume')
+
+    @property
+    def tenant(self):
```

Comment:
**REQUIRED**: Since this does not return the tenant object but just the name, ``tenant_name`` will be a more appropriate name for this property.